spring-integration-file/src/main/java/org/springframework/integration/file/splitter/FileSplitter.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2015 the original author or authors
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ */
+
+package org.springframework.integration.file.splitter;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import org.springframework.integration.splitter.AbstractMessageSplitter;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessagingException;
+
+/**
+ * The {@link AbstractMessageSplitter} implementation to split the {@link File}
+ * Message payload to lines.
+ * <p>
+ * With {@code iterator = true} (defaults to {@code false}) this class produces an {@link Iterator}
+ * to process file lines on demand from {@link Iterator#next}.
+ * Otherwise the {@link List} of all lines is returned to the to further
+ * {@link AbstractMessageSplitter#handleRequestMessage} process.
+ * <p>
+ *  Can accept {@link String} as file path, {@link File}, {@link Reader} or {@link InputStream}
+ *  as payload type.
+ *  All other types are ignored and returned to the {@link AbstractMessageSplitter) as is.
+ *
+ * @author Artem Bilan
+ * @since 4.1.2
+ */
+public class FileSplitter extends AbstractMessageSplitter {
+
+	private boolean iterator;
+
+	public void setIterator(boolean iterator) {
+		this.iterator = iterator;
+	}
```

Comment:
Let's make it `final` and a constructor arg.

I also think it should be `true` by default.

To be honest, I see little use for the `false` case since we'll load the entire file into memory and then `produceOutput` iterates over the collection. While there might be a corner case where that is desired, I suspect the iterator version would be most useful.